pvlib/ivtools.py

```diff
@@ -262,6 +264,163 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
+    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88,
+    2006.
+
+    [2] John A Dufﬁe ,William A Beckman, "Solar Engineering of Thermal
+    Processes", Wiley, 2013
+    """
+    # Constants
+    k = constants.value('Boltzmann constant in eV/K')
+    Tref = temp_ref + 273.15  # [K]
+
+    def pv_fct(params, specs):
+        """Evaluates the systems of equations used to solve for the single
+        diode equation parameters.
+        To avoid the confusion in names with variables of container
+        function the '_' of the variables were removed.
+        """
+        # six input known variables
+        Isc, Voc, Imp, Vmp, betaoc, alphasc = specs
+
+        # five parameters vector to find
+        IL, Io, a, Rsh, Rs = params
+
+        # five equation vector
+        y = [0, 0, 0, 0, 0]
+
+        # 1st equation - short-circuit - eq(3) in [1]
+        y[0] = Isc - IL + Io*np.expm1(Isc*Rs/a) + Isc*Rs/Rsh
+        # 2nd equation - open-circuit Tref - eq(4) in [1]
+        y[1] = -IL + Io*np.expm1(Voc/a) + Voc/Rsh
+        # 3rd equation - Imp & Vmp - eq(5) in [1]
+        y[2] = Imp - IL + Io*np.expm1((Vmp+Imp*Rs)/a) + \
+            (Vmp+Imp*Rs)/Rsh
+        # 4th equation - Pmp derivated=0 - eq23.2.6 in [2]
+        # caution: eq(6) in [1] has a sign error
+        y[3] = Imp - Vmp * ((Io/a)*np.exp((Vmp+Imp*Rs)/a) + 1.0/Rsh) / \
+            (1.0 + (Io*Rs/a)*np.exp((Vmp+Imp*Rs)/a) + Rs/Rsh)
+        # 5th equation - open-circuit T2 - eq (4) at temperature T2 in [1]
+        T2 = Tref + 2
+        Voc2 = (T2 - Tref)*betaoc + Voc  # eq (7) in [1]
+        a2 = a*T2/Tref  # eq (8) in [1]
+        IL2 = IL + alphasc*(T2-Tref)  # eq (11) in [1]
+        Eg2 = EgRef*(1 + dEgdT*(T2-Tref))  # eq (10) in [1]
+        Io2 = Io * (T2/Tref)**3 * np.exp(1/k * (EgRef/Tref-Eg2/T2))  # eq (9)
+        y[4] = -IL2 + Io2*np.expm1(Voc2/a2) + Voc2/Rsh  # eq (4) at T2
+
+        return y
+
+    # initial guesses of variables for computing convergence:
+    # Values are taken from [2], p753
+    Rsh_i = 100.0
+    a_i = 1.5*k*Tref*cells_in_series
+    IL_i = i_sc
+    Io_i = i_sc * np.exp(-v_oc/a_i)
+    Rs_i = (a_i*np.log1p((IL_i-i_mp)/Io_i) - v_mp)/i_mp
+    # params_i : initial values vector
+    params_i = np.array([IL_i, Io_i, a_i, Rsh_i, Rs_i])
+
+    # specs of module
+    specs = np.array([i_sc, v_oc, i_mp, v_mp, beta_voc, alpha_sc])
+
+    # computing
+    result = root(pv_fct, x0=params_i, args=specs, method=solver_method)
+
+    if result.success:
+        sdm_params = result.x
+    else:
+        raise RuntimeError('Parameter estimation failed')
+
+    # results
+    return {'I_L_ref': sdm_params[0],
+            'I_o_ref': sdm_params[1],
+            'a_ref': sdm_params[2],
+            'R_sh_ref': sdm_params[3],
+            'R_s': sdm_params[4],
+            'alpha_sc': alpha_sc,
+            'EgRef': EgRef,
+            'dEgdT': dEgdT,
+            'irrad_ref': irrad_ref,
+            'temp_ref': temp_ref}
```

Comment:
I personally like to be able to inspect the full `result` object from the solver (possibly as a second output in a returned tuple), but I also understand if the majority would rather omit it from the return interface.

Reply:
Ok. 
Then do you think it is worth testing the result object in the test ? I feel like some attribute of the complete OptimizeResult object can easily change (like nit the number of iteration, or nfev) if scipy is improved. That's why I would remove the optimizeresult from the dict output in test_fit_sdm_desoto() before testing the equivalence with expected result.

Reply:
I don't think it is worth testing the content of `result` in the CI tests. 

However, your question led me to look at the default settings for `solver_method in 'lm', 'hybr '`. The default settings terminate based on relative change of the solution, so it is possible that the root finder stalls in some flat portion of the objective surface rather than finding the minimum. We could build our own function to test the quality of the solution that `root` finds, but let's do that in a follow-on PR.